dokka-runners/dokka-gradle-plugin/src/testFixtures/kotlin/GradleTestKitUtils.kt

```diff
@@ -164,11 +162,32 @@ class GradleProjectTest(
             .withProjectDir(projectDir.toFile())
             .writeGradleProperties(gradleProperties)
 
+    /**
+     * Update all repositories in any `settings.gradle(.kts)` file.
+     *
+     * All `repositories {}` will be modified to add the dev maven repos, [mavenRepositories].
+     */
+    fun updateSettingsRepositories() {
+        projectDir.walk()
+            .filter { it.name == "settings.gradle.kts" || it.name == "settings.gradle" }
+            .forEach { p ->
+                val repoLine = p.useLines { it.firstOrNull { l -> l.trim() == "repositories {" } }
```

Comment:
This implementation looks somewhat fragile because it relies on seeking `repositories {` substring. Also what is the use-case for calling this function, but then doing nothing if the repositories are not specified in the test project?

Can we simplify this implementation to:
* Find and assert `settings.gradle(.kts)` file exists
* Inject `pluginManagement` and `dependencyResolutionManagement` blocks with necessary repositories

Reply:
> This implementation looks somewhat fragile because it relies on seeking `repositories {` substring. Also what is the use-case for calling this function, but then doing nothing if the repositories are not specified in the test project?

If the repositories aren't correctly updated, then I would expect that the test will fail. The version of DGP is not published to a remote repository, and is only available in a local directory.

> Can we simplify this implementation to:
> 
> * Find and assert `settings.gradle(.kts)` file exists

Some of the example projects (e.g. the composite build example) have multiple `settings.gradle`s, so they need to be discovered and updated dynamically. We could try to manually specify this per project, but since the tests should fail without a correct repository, doing an automatic find/replace seems fine to me.

> * Inject `pluginManagement` and `dependencyResolutionManagement` blocks with necessary repositories

Do you mean appending multiple `pluginManagement {}` and `dependencyResolutionManagement {}` blocks at the end of the file?

That could work for `dependencyResolutionManagement {}`, because Gradle will allow repeated blocks, but for `pluginManagement {}` Gradle will fail if there are multiple blocks, so `pluginManagement{ repositories {} }` has to be updated in-place.

Maybe an init plugin could help? But I'm not sure if that could work with the composite build example.

Reply:
My main argument for not doing this substring search/replace is that I find reasoning about such code cumbersome if the behavior doesn't match my expectation about what this code should be doing. If you want to keep the substring search/replace, I would advise to instead use an explicit marker:
```
repositories {
  <repositories_marker>
}
```
which will explicitly fail to evaluate if it wasn't processed

> pluginManagement{ repositories {} } has to be updated in-place.

Can we avoid specifying `pluginManagement` to avoid having to patch it later?

> work with the composite build example

Have you tried it out? I would expect init scripts to apply to all project hierarchies in the composite build